HW2/P5/P5.txt

```diff
@@ -0,0 +1,75 @@
+Part 1:
+    Performance with 1 thread:
+        9.91139467839 simulation frames per second
+        9.90245583881 simulation frames per second
+        9.88816853461 simulation frames per second
+        9.88914771816 simulation frames per second
+        9.87838735357 simulation frames per second
+        Average: 9.894 simulation frames per second
+    
+    Performance with 4 threads:
+        22.5982554161 simulation frames per second
+        22.5825598174 simulation frames per second
+        22.5820734806 simulation frames per second
+        22.5835325537 simulation frames per second
+        22.5739304532 simulation frames per second
+        Average: 22.584 simulation frames per second
+    
+    On average, we achieve more than twice as many simulations per second using 4 threads instead of 1.
+    
+Part 2 Griding:
+    Performance with 1 thread:
+        564.017863588 simulation frames per second
+        558.31588387 simulation frames per second
+        560.870008305 simulation frames per second
+        559.605749717 simulation frames per second
+        561.826908435 simulation frames per second
+        Average: 560.927 simulation frames per second
+    
+    Performance with 4 threads:
+        1605.25901881 simulation frames per second
+        1616.10606007 simulation frames per second
+        1647.96407723 simulation frames per second
+        1634.40684691 simulation frames per second
+        1630.7097212 simulation frames per second
+        Average: 1626.889 simulation frames per second
+
+    On average, we achieve close to three times as many simulations per second using 4 threads instead of 1.  The addition of griding improves performance substantially from part 1.
```

Comment:
Griding leads to the 50x boost for one thread.  Increased overhead leads to the 72x increase for 4 threads (better performance, but doesn't scale at 2x speed for 2x as many threads).